corehq/apps/domain/models.py

```diff
@@ -534,7 +518,7 @@ def __unicode__(self):
         return self.name
 
     @classmethod
-    @quickcache(['name'], timeout=30*60, helper_class=_StrictQuickCache)
+    @quickcache(['name'], timeout=30*60, helper_class=skippable_quick_cache('strict'))
```

Comment:
almost seems like you'd want the interface to be

`@skippable_quickcache(['name'], timeout=30*60, skip_arg='strict')`

Then `skippable_quickcache` below would return `quickcache(*args, helper_class=CustomSkippableQuickCache(skip_arg), **kwargs)`.

If you like this suggestion, I'll leave this PR open; otherwise this looks good to go (and a great improvement!)